src/intptrcast.rs

```diff
@@ -105,15 +105,16 @@ impl<'mir, 'tcx> GlobalStateInner {
     pub fn expose_ptr(
         ecx: &mut MiriInterpCx<'mir, 'tcx>,
         alloc_id: AllocId,
-        sb: SbTag,
+        tag: BorTag,
     ) -> InterpResult<'tcx> {
         let global_state = ecx.machine.intptrcast.get_mut();
         // In strict mode, we don't need this, so we can save some cycles by not tracking it.
         if global_state.provenance_mode != ProvenanceMode::Strict {
             trace!("Exposing allocation id {alloc_id:?}");
             global_state.exposed.insert(alloc_id);
-            if ecx.machine.stacked_borrows.is_some() {
-                ecx.expose_tag(alloc_id, sb)?;
+            match ecx.machine.borrow_tracker_method {
+                BorrowTrackerMethod::StackedBorrows => ecx.sb_expose_tag(alloc_id, tag)?,
+                BorrowTrackerMethod::Off => (),
```

Comment:
As I said last time, there shouldn't be any `match` on this outside the borrow_tracker module.

Reply:
Oops sorry I missed that one